crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -598,3 +633,185 @@ fn is_expression_a_promise(expr: &AnyJsExpression, model: &SemanticModel) -> boo
 
     true
 }
+
+/// Traverses up the syntax tree to find the class declaration and checks if a method is a promise.
+///
+/// This function traverses up the syntax tree from the given `JsThisExpression` to find the nearest
+/// class declaration. It then searches for a method or property in the class that matches the provided
+/// `target_name`. If a matching member is found, it checks if the member is a promise.
+///
+/// # Arguments
+///
+/// * `js_this_expression` - A `JsThisExpression` representing the `this` keyword in the syntax tree.
+/// * `target_name` - The name of the method or property to search for.
+/// * `model` - A reference to the `SemanticModel` used for resolving bindings.
+///
+/// # Returns
+///
+/// * `Some(true)` if the class member is a promise.
+/// * `Some(false)` if the class member is not a promise.
+/// * `None` if there is an error in the process or if the class member is not found.
+///
+/// # Examples
+///
+/// Example TypeScript code that would return `Some(true)`:
+/// ```typescript
+/// class Api {
+///   async returnsPromise(): Promise<string> {
+///     return 'value';
+///   }
+///   async someMethod() {
+///     this.returnsPromise();
+///   }
+/// }
+/// ```
+///
+/// Example TypeScript code that would return `Some(false)`:
+/// ```typescript
+/// class Api {
+///   returnsString(){
+///     return 'value';
+///   }
+///   async someMethod() {
+///     this.returnsString();
+///   }
+/// }
+/// ```
+fn check_this_expression(
+    js_this_expression: &JsThisExpression,
+    target_name: &str,
+    model: &SemanticModel,
+) -> Option<bool> {
+    js_this_expression
+        .syntax()
+        .ancestors()
+        .skip(1)
+        .find_map(|ancestor| {
+            if ancestor.kind() == JsSyntaxKind::JS_CLASS_DECLARATION {
+                let class_decl = JsClassDeclaration::cast(ancestor)?;
+                return find_and_check_class_member(&class_decl, target_name, model);
+            }
+            None
+        })
+}
+
+fn check_this_expression_in_static_member_expression(
+    static_member_expr: &JsStaticMemberExpression,
+    model: &SemanticModel,
+) -> Option<bool> {
+    let any_js_expression = static_member_expr.object().ok()?;
+    let js_this_expression = any_js_expression.as_js_this_expression()?;
+    let js_name = static_member_expr.member().ok()?;
+    let value_token = js_name.value_token().ok()?;
+    if !check_this_expression(js_this_expression, &value_token.to_string(), model)? {
```

Comment:
Most of the times, you don't want to use `to_string`. You want to use methods like `text_trimmed()` or `inner_string_text`

`to_string` **allocates** a string.

Reply:
I replaced it with `text_trimmed`